File: dnf.py
def sort_colors(nums):

    colors = {
        0: 0,
        1: 0, 
        2: 0
    }

    for num in nums:
        colors[num] += 1

    print(colors)

    index = 0
    for color in range(3):
        for _ in range(colors[color]):
            nums[index] = color
            index += 1

    return nums

# Example usage
arr = [2, 0, 2, 1, 0, 1, 1, 0, 2, 1, 0, 2]
print("Original array:", arr)
sorted_arr = sort_colors(arr)
print("Sorted array:", sorted_arr)

File: dnf.c
#include <stdio.h>


void sortColors(int* nums, int numSize) {
    int count[3] = {0};

    for (int i=0; i < numSize; i++) {
        count[nums[i]]++;
    }

    int i = 0;
    for (int color = 0; color < 3; color++) {
        for (int j = 0; j < count[color]; j++) {
            nums[i++] = color;
        }
    }
}

void printArray(int* nums, int numsSize) {
    for (int i = 0; i < numsSize; i++) {
        printf("%d ", nums[i]);
    }
    printf("\n");
}

int main() {
    int arr[] = {2, 0, 1, 1, 0, 2, 1, 0, 2};
    int size = sizeof(arr) / sizeof(arr[0]);
    
    printf("Original array: ");
    printArray(arr, size);
    
    sortColors(arr, size);
    
    printf("Sorted array: ");
    printArray(arr, size);
    
    return 0;
}
